examples/summarizePatches.py

```diff
@@ -0,0 +1,179 @@
+    ra, dec = zip(*corners)
+
+    # make Box2D with WCS coordinates
+    minPoint = afwGeom.Point2D(min(ra), min(dec))
+    maxPoint = afwGeom.Point2D(max(ra), max(dec))
+    wcsBBox = afwGeom.Box2D(minPoint, maxPoint)
+    return wcsBBox
+
+def main(rootDir, visits=None, ccdKey='ccd', saveOverview=False, saveSummaries=None):
+    """Find which ccds overlap which patch and create summaries of visits organized by patch."""
+    butler = dafPersist.Butler(rootDir)
+    camera = butler.get('camera')
+    skymap = butler.get('deepCoadd_skyMap')
+
+    # open files for overview and summaries if needed
+    if saveOverview:
+        overview = open("patchesOverview", 'w')
+    if saveSummaries is not None:
+        expandedFileName = "expanded_" + saveSummaries
+        condensedFileName = "condensed_" + saveSummaries
+        expandedSummary = open(expandedFileName, 'w')
+        condensedSummary = open(condensedFileName, 'w')
+
+    # Default to all visits if none are specified
+    if visits is None:
+        visits = butler.queryMetadata('calexp_md', 'visit')
+
+    # Make BBoxes of each ccd in each visit
+    skymapBBoxes = {}
+    for visit in visits:
+        for ccd in camera:
+            ccdId = int(ccd.getSerial())
+            bbox = ccd.getBBox()
+            dataId = {'visit': visit, ccdKey: ccdId}
+            try:
+                md = butler.get('calexp_md', dataId)
+                wcs = afwImage.makeWcs(md)
+                ccdBBox = wcsBBox(bbox, wcs)
+                skymapBBoxes[(visit, ccdId)] = ccdBBox
+            except:
+                pass
+
+    # Run through patches and find overlaps with ccds
+    for tract in skymap:
+        for patch in tract:
+            visitsToCcds = {}
+            filtersToVisits = {}
+            numOverlaps = 0
+            patchId = patch.getIndex()
+            patchBBox = wcsBBox(patch.getInnerBBox(), tract.getWcs())
+
+            # Find which ccds the patch overlaps (if any)
+            for skymapId in skymapBBoxes:
+                if patchBBox.overlaps(skymapBBoxes[skymapId]):
+                    visitId, ccdId = skymapId
+                    if visitId not in visitsToCcds:
+                        visitsToCcds[visitId] = [ccdId]
+                        visitFilter = butler.queryMetadata('calexp_md', 'filter', visit=visitId)[0]
+                        if visitFilter not in filtersToVisits:
+                            filtersToVisits[visitFilter] = []
+                        filtersToVisits[visitFilter] += [visitId]
+                        numOverlaps += 1
+                    else:
+                        visitsToCcds[visitId] += [ccdId]
+                        numOverlaps += 1
+
+            # If there was an overlap, print and save information
+            if numOverlaps != 0:
+                print("patch=%r:    Found %r overlapping ccds." % (patchId, numOverlaps))
+
+                if saveOverview:
+                    overview.write("patch=" + str(patchId) + "\n")
```

Comment:
I would probably have written all of these with format strings instead of concatenation. Mostly a stylistic thing, but I think it's a little less cluttered.